Fix flakiness on enrollment integration test and rename it

AndersonQ opened the pull request:
## What does this PR do?

Fix flakiness on enrollment integration test and rename it so it better reflects what is tested by it.

I continually run the test for 20h and it did not fail.

## Why is it important?

The test was flaky and the name was misleading. It could lead one to believe the test was just the enrollement, when it actually tests the enrollment and the monitoring is shipping logs.

## Checklist

<!-- Mandatory
Add a checklist of things that are required to be reviewed in order to have the PR approved

List here all the items you have verified BEFORE sending this PR. Please DO NOT remove any item, striking through those that do not apply. (Just in case, strikethrough uses two tildes. ~~Scratch this.~~)
-->

- [x] My code follows the style guidelines of this project
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `./changelog/fragments` using the [changelog tool](https://github.com/elastic/elastic-agent#changelog)~~
- ~~[ ] I have added an integration test or an E2E test~~

## How to test this PR locally

Run it continuously, it should not fail.

```sh
AGENT_VERSION=8.10.0-SNAPSHOT AGENT_KEEP_INSTALLED=true TEST_RUN_UNTIL_FAILURE=true mage integration:single TestMonitoringLogsShipped
```

I run it for 20h, no issues.

## Related issues

- Closes #3081 

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...

elasticmachine commented:
Pinging @elastic/elastic-agent (Team:Elastic-Agent)

cmacknz reviewed (changes requested):
This test is intentionally testing that we can ship Beat logs to Fleet.

Otherwise we have no test coverage for the monitoring Filebeat instance and its configuration. See https://github.com/elastic/elastic-agent/issues/2402.

The test name is just not specific enough to communicate this.

If we do have this functionality covered somewhere else we can keep the deletion, but we can't just delete it completely.

ycombinator commented on testing/integration/monitoring_logs_test.go:
```diff
@@ -15,49 +15,34 @@ import (
-	info := runner.requirementsInfo
-
-	kibClient := info.KibanaClient
+	t.Logf("In SetupSuite")
```
I think this line can be removed now?

ycombinator commented on testing/integration/monitoring_logs_test.go:
```diff
@@ -92,15 +79,15 @@ func (runner *EnrollRunner) TestEnroll() {
 
-	t.Logf("sleeping for one minute...")
-	time.Sleep(time.Second * 60)
+	AssertConnectedToFleet(t, agentFixture)
```
Now that this helper function is being used by multiple tests, could we move it's definition out of `proxy_url_test.go` and into somewhere under https://github.com/elastic/elastic-agent/tree/main/pkg/testing/tools?

ycombinator commented on testing/integration/monitoring_logs_test.go:
```diff
@@ -111,16 +98,16 @@ func (runner *EnrollRunner) TestEnroll() {
-	for _, doc := range docs.Hits.Hits {
-		t.Logf("%#v", doc.Source)
+	// Stage 4: make sure all components are health
+	t.Log("Making sure all components are health")
```
Typo.

```suggestion
	t.Log("Making sure all components are healthy")
```

ycombinator commented on testing/integration/monitoring_logs_test.go:
```diff
@@ -111,16 +98,16 @@ func (runner *EnrollRunner) TestEnroll() {
+	t.Log("Making sure all components are health")
+	status, err := agentFixture.ExecStatus(ctx)
+	require.NoError(t, err,
+		"could not get agent status to verify all components are health")
```
Typo.
```suggestion
		"could not get agent status to verify all components are healthy")
```

ycombinator commented on testing/integration/proxy_url_test.go:
```diff
@@ -392,19 +392,29 @@ func TestProxyURL_RemoveProxyFromThePolicy(t *testing.T) {
 	p.assertConnectedFleet(t)
 }
 
+// assertConnectedFleet is deprecated, use AssertConnectedToFleet instead
+// Deprecated.
 func (p *ProxyURL) assertConnectedFleet(t *testing.T) {
+	AssertConnectedToFleet(t, p.fixture)
+}
+
```
Why not just remove this function and replace all its calls with `AssertConnectedToFleet`? From what I can tell, this method is only being called from within `proxy_url_test.go`. Is it to keep the size of this PR small? In that case I'd do this cleanup in an immediate follow up PR so we don't forget about it.

AndersonQ replied:
it was temporary while I was checking exactly what the test was intended to do and if my fix worked. I'm fixing it all now

ycombinator commented on testing/integration/proxy_url_test.go:
```diff
@@ -392,19 +392,29 @@ func TestProxyURL_RemoveProxyFromThePolicy(t *testing.T) {
-		cproto.State_HEALTHY, cproto.State(agentStatus.FleetState), agentStatus) {
+		cproto.State_HEALTHY, cproto.State(agentStatus.FleetState), agentStatus)
+
+	if !connected {
 		if err != nil {
```
Nit: could be simplified.
```suggestion
	if !connected && err != nil {
```

AndersonQ commented:
@cmacknz I restored the test, changed its name to better reflect what it tests and fixed the issue. Could you re-review it?

@ycombinator, I applied your suggestions

ycombinator reviewed (approved):
LGTM.

cmacknz commented on testing/integration/monitoring_logs_test.go:
```diff
@@ -15,49 +15,34 @@ import (
 	info := define.Require(t, define.Requirements{
 		OS: []define.OS{
-			{Type: define.Linux},
+			{Type: define.Linux,
```
FYI the framework now supports doing this from the command line https://github.com/elastic/elastic-agent/blob/main/docs/test-framework-dev-guide.md#selecting-specific-platform

cmacknz commented on testing/integration/monitoring_logs_test.go:
```diff
@@ -111,21 +98,22 @@ func (runner *EnrollRunner) TestEnroll() {
-	// Stage 4: make sure we have no errors
-	t.Log("Making sure there are no error logs")
-	docs = findESDocs(t, func() (tools.Documents, error) {
-		return tools.CheckForErrorsInLogs(info.ESClient, info.Namespace, []string{})
```
What were the error logs this was failing on? Should they exist or are we just ignoring a bug with this? It may be better to filter out known false positives than to remove this. 

There have been a lot of problems in the logs that are hidden because we have no automation that can detect them, this is one of the few places that can do that for us.

I am worried we are "fixing" the test by making it a worse test.

AndersonQ replied:
It's not the case here. The error logs were from the agent trying to connect to the default ES, `localhost:9200`. Besides using the health of the agent and its components makes the test better, the test won't be affected by changes in the log messages and transient/recoverable errors.

This tests want to ensure the agent enrolls and the monitoring works. If any issue happens and the agent is capable of recovering and reaches the final desired state, it's not an issue. It shows the "self-healing" strategies we have work. Also taking into consideration it's an e2e test we cannot control every single aspect of the process, what we need to ensure is the final state, the agent is enrolled and monitoring is working, what is being done by this test.

AndersonQ commented:
@cmacknz could you review this PR again?
